src/common/arrstr.cpp

```diff
@@ -721,3 +730,212 @@ wxArrayString wxSplit(const wxString& str, const wxChar sep, const wxChar escape
+
+
+    wxStringFragment GetFragment(wxString& text)
+    {
+        static const wxRegEx reSpaceOrPunct(wxS("^([[:space:]]|[[:punct:]])+"));
+        static const wxRegEx reDigit(wxS("^[[:digit:]]+"));
+        static const wxRegEx reLetterOrSymbol("^[^[:space:]|[:punct:]|[:digit:]]+");
+
+        wxStringFragment fragment;
+        size_t           length = wxString::npos;
+
+        if ( text.empty() )
+            return fragment;
+
+        // In attempt to minimize the number of wxRegEx.Matches() calls,
+        // try to do them from the most expected to the least expected
+        // string fragment type.
+        if ( reLetterOrSymbol.Matches(text) )
+        {
+            if ( reLetterOrSymbol.GetMatch(NULL, &length) )
+            {
+                fragment.type = wxStringFragment::LetterOrSymbol;
+                fragment.text = text.Left(length);
+            }
+        }
+        else if ( reDigit.Matches(text) )
+        {
+            if ( reDigit.GetMatch(NULL, &length) )
+            {
+                fragment.type = wxStringFragment::Digit;
+
+                // The number may be too big to fit into a wxUint64,
+                // so take at most its first 19 digits, the rest of
+                // the number will be processed as the next string fragment.
+                // Not perfect, but I could not think of a better simple solution.
+                length = wxMin(length, 19);
+                fragment.text = text.Left(length);
+                fragment.text.ToULongLong(&fragment.value);
+            }
+        }
+        else if ( reSpaceOrPunct.Matches(text) )
+        {
+            if ( reSpaceOrPunct.GetMatch(NULL, &length) )
+            {
+                fragment.type = wxStringFragment::SpaceOrPunct;
+                fragment.text = text.Left(length);
+            }
+        } else
+        {
+            wxFAIL_MSG("text doesn't start with any recognizable fragment type");
+            return fragment;
+        }
+
+        text.erase(0, length);
+        return fragment;
+    }
+
+    int CompareFragmentNatural(const wxStringFragment& lhs, const wxStringFragment& rhs)
+    {
+        int result = 1;
+
+        if ( lhs.type == wxStringFragment::Empty )
+        {
+            if ( rhs.type == wxStringFragment::Empty )
+                result = 0;
+            else
+                result = -1;
+        }
+        else if ( lhs.type == wxStringFragment::SpaceOrPunct )
+        {
+            switch ( rhs.type )
+            {
+                case wxStringFragment::Empty:
+                    result = 1;
+                case wxStringFragment::SpaceOrPunct:
+                    result = wxStrcoll_String(lhs.text, rhs.text);
+                    break;
+                case wxStringFragment::Digit:
+                case wxStringFragment::LetterOrSymbol:
+                    result = -1;
+                    break;
+                default:
+                    wxFAIL_MSG("Unknown string fragment type");
```

Comment:
For the same exhaustiveness reasons I'd remove this because compiler wouldn't warn about missing case if we ever add another fragment type with the `default` present.

(also below)

Reply:
Will do.